skills/csharp/todoskill/Dialogs/MainDialog.cs

```diff
@@ -385,5 +443,34 @@ private void InitializeConfig(ToDoSkillState state)
                 }
             }
         }
+
+        private async Task DigestActionInput(DialogContext dc, object actionInput)
```

Comment:
If only have two like ToDoInfo, ListInfo, could use two functions to specify the class to use.
Or use a base class / interface to unify the function.

For current code, it will always end in first try because json will just fill the attributes

Reply:
Thanks. I delete the second part, because the first part can fill all properties already.

Reply:
No, you misunderstand the mechanism of json. In this way, use could use every action with ToDoInfo and fill all state variables. It may be not harm this time but it is a bad design.
Anyway, pay attention next time.